mailutils.php

```diff
@@ -82,3 +83,65 @@ function mailutils_civicrm_navigationMenu(&$menu) {
   ));
   _mailutils_civix_navigationMenu($menu);
 }
+
+/**
+ * Implements hook_coreResourceList
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_coreResourceList
+ */
+function mailutils_civicrm_coreResourceList(&$list, $region) {
+  CRM_Core_Resources::singleton()->addStyleFile('mailutils', 'css/mailutils.css', 0, 'page-header');
+}
+
+/**
+ * Implements hook_civicrm_buildForm().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_buildForm
+ */
+function mailutils_civicrm_buildForm($form_name, &$form) {
+  switch ($form_name) {
+    case 'CRM_Activity_Form_ActivityView': {
+      $activity_template_data = $form->getTemplateVars('values');
+      $activity_id = $activity_template_data['id'];
+      $email_display = CRM_Mailutils_Utils_MessageRenderer::render($activity_id);
+
+      if (is_null($email_display)) return;
+
+      $activity_template_data['details'] = $email_display;
+      $form->assign('values', $activity_template_data);
+
+      break;
+    }
+
+    case 'CRM_Case_Form_ActivityView': {
+      $activity_template_data = $form->getTemplateVars('report');
+      $activity_id = $form->getTemplateVars('activityID');
+      $email_display = CRM_Mailutils_Utils_MessageRenderer::render($activity_id);
+
+      if (is_null($email_display)) return;
+
+      $field_index = array_find_key(
```

Comment:
`array_find_key()` is PHP 8.4+ only (we're targeting 8.1 for standalone, though this change could be deployed even on 5.69 with PHP 7.4).